### PR TITLE
Improve releasing instructions related to Nexus.

### PR DESCRIPTION
**Pull request by k-wall**
### Type of change

_Select the type of your PR_

- Documentation

### Description

Improve releasing instructions related to Nexus.

Why: this change was prompted by these comments https://github.com/kroxylicious/kroxylicious-junit5-extension/pull/199#discussion_r1353287860 

### Additional Context

_Why are you making this pull request?_

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Write tests
- [ ] Make sure all tests pass
- [x] Update documentation
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md

### CODE REVIEW

File: RELEASING.md [hunk 1]
```diff
@@ -21,8 +21,50 @@ the release script to actually perform the release and prepare main for the next
...
+The script will push the Maven artefacts to a Nexus **staging repository** and it will open a PR that will commit the release to Git.
+
+The next steps are:
+1.  Validate the staged artefacts.
+2.  If the staging artefacts are **good**, release the stage repository.  If they are not, **drop** the stage repository and start the release over.
+3.  Merge the PR.
+
+These are described below.
+
+### Validating the staged artefacts
```
Comment (k-wall) [inline RELEASING.md:33]: We could have a Git Action for this bit.

Comment (SamBarker) [reply RELEASING.md:33]: Yeah that would be very cool, but for another day.

File: RELEASING.md [hunk 2]
```diff
@@ -21,8 +21,50 @@ the release script to actually perform the release and prepare main for the next
...
+
+You can validate the staged artefacts by having a test application (we'll call it `T`) that uses kroxylicious-junit5-extension use the Maven artefacts by making
```
Comment (SamBarker) [inline RELEASING.md:35]: Should we just have a stub project in GitHub that we use for validating the release?

I had loosely hoped at one stage to use the Sample plug-in for the task but I think I had missed the close step being required for publicly accessible staging repo URL

File: RELEASING.md [hunk 3]
```diff
@@ -21,8 +21,50 @@ the release script to actually perform the release and prepare main for the next
...
+
+### Dropping the stage repository
+
+If the validation step fails and the maven artefacts are deemed bad, drop the staging repository.
+
+Use the [Drop](https://help.sonatype.com/repomanager2/staging-releases/managing-staging-repositories) to drop the stage repository.
+
+In this case the PR should be closed without merging.
+
+Once the issue is resolved, repeat the release, bumping the release micro by 1.
```
Comment (SamBarker) [inline RELEASING.md:65]: `X.Y.Z` release would not have happened is there any reason to bump Z by 1? We can just fix the issue and re-run the release process can't we?

**Review by SamBarker — APPROVED** [review]
Great write up, thanks  @k-wall

File: RELEASING.md (hunk 2 above)
Comment (k-wall) [reply RELEASING.md:35]: Stub projects (one that consumes the junit ext and one that consumes the filter api) would make sense I think, but I haven't given it too much thought.

File: RELEASING.md (hunk 3 above)
Comment (k-wall) [reply RELEASING.md:65]: The tag has already been pushed to the origin. The choice is either bump the micro or be prepared to force the tag.  If we go the force tag route, I'd write the script to be behave conservatively and only force the tag if given an extra argument (which would be surfaced at the Git Action level and default to false).

Comment (k-wall) [reply RELEASING.md:65]: I think we can decide later.

**Pull request merged**